packages/modal/src/useModal.ts

```diff
@@ -9,18 +9,37 @@ import { useState } from 'react';
 import { composeEventHandlers, generateId, KEY_CODES } from '@zendeskgarden/container-utilities';
 import { useFocusJail } from '@zendeskgarden/container-focusjail';
 
-export function useModal({ onClose, modalRef, id: _id, focusOnMount, environment } = {}) {
+export interface IUseModalProps {
+  onClose?: (event: Event) => any;
+  modalRef: React.RefObject<HTMLElement>;
+  id?: string;
+  focusOnMount?: boolean;
+  environment?: Document;
+}
+
+export interface IUseModalReturnValue {
+  getBackdropProps: <T>(options?: T) => T & React.HTMLProps<any>;
+  getModalProps: <T>(options?: T) => T & React.HTMLProps<any>;
+  getTitleProps: <T>(options?: T) => T & React.HTMLProps<any>;
+  getContentProps: <T>(options?: T) => T & React.HTMLProps<any>;
+  getCloseProps: <T>(options?: T) => T & React.ButtonHTMLAttributes<any>;
+  closeModal?: <T>(event: T & Event) => void;
```

Comment:
This shouldn't need a generic. Let's try `closeModal?: (event: MouseEvent | KeyboardEvent) => void`

Reply:
That's right, this does not need to be a generic function. I've removed the generic and updated argument type to `any`.